### PR TITLE
DAOS-2651 rebuild: abort rebuild once scan fails

### PR DESCRIPTION
**Pull request by wangdi1**
Let's return the status of scan bcast, and abort
current rebuild immediately, and rebuild all targets
in the next rebuild.

Signed-off-by: Wang Di <di.wang@intel.com>

### CODE REVIEW

File: src/tests/suite/daos_rebuild.c [hunk 1]
```diff
@@ -1414,22 +1432,34 @@ rebuild_master_change_during_rebuild(void **state)
...
+	if (arg->myrank == 0)
+		daos_mgmt_set_params(arg->group, -1, DSS_KEY_FAIL_LOC,
+				     DAOS_REBUILD_TGT_REBUILD_HANG, 0, NULL);
+	MPI_Barrier(MPI_COMM_WORLD);
+	new_arg->rebuild_cb = rebuild_change_leader_cb;
 
-	rebuild_single_pool_rank(arg, ranks_to_kill[0]);
+	rebuild_single_pool_rank(new_arg, ranks_to_kill[0]);
 
 	arg->rebuild_cb = NULL;
```
Comment (gnailzenh) [inline src/tests/suite/daos_rebuild.c]: unnecessary right?

File: src/tests/suite/daos_rebuild.c [hunk 2]
```diff
@@ -1380,31 +1385,44 @@ rebuild_master_change_during_scan(void **state)
...
 		daos_mgmt_set_params(arg->group, -1, DSS_KEY_FAIL_LOC,
 				     DAOS_REBUILD_TGT_SCAN_HANG, 0, NULL);
 	MPI_Barrier(MPI_COMM_WORLD);
-	arg->rebuild_cb = rebuild_change_leader_cb;
+	new_arg->rebuild_cb = rebuild_change_leader_cb;
 
-	rebuild_single_pool_rank(arg, ranks_to_kill[0]);
+	rebuild_single_pool_rank(new_arg, ranks_to_kill[0]);
 
 	arg->rebuild_cb = NULL;
```
Comment (gnailzenh) [inline src/tests/suite/daos_rebuild.c]: unnecessary right?

File: src/tests/suite/daos_rebuild.c [hunk 3]
```diff
@@ -1047,6 +1049,7 @@ rebuild_tgt_start_fail(void **state)
 	rebuild_io_validate(arg, oids, OBJ_NR, true);
 
 	rebuild_add_back_tgts(&arg, 1, ranks_to_kill, NULL, 1);
+	rebuild_add_back_tgts(&arg, 1, &exclude_rank, NULL, 1);
```
[the remark this answers is not in the document]
Comment (wangdi1) [reply src/tests/suite/daos_rebuild.c]: this test will fake rank 0 failure, which will cause it to be excluded during the test. so we have to add it back.

[the remark this answers is not in the document]
Comment (liuxuezhao) [reply src/tests/suite/daos_rebuild.c]: i see, in rebuild_tgt_scan_handler the DAOS_REBUILD_TGT_START_FAIL will fail it, and aggregate to rso_ranks_list and on bcast initiate node will exclude it... maybe some comments is helpful.

File: src/tests/suite/daos_rebuild.c [hunk 4]
```diff
@@ -34,12 +34,13 @@
 #include <daos/mgmt.h>
 #include <daos/container.h>
 
-#define KEY_NR		1000
+#define KEY_NR		100
```
Comment (jolivier23) [inline src/tests/suite/daos_rebuild.c]: should this be tunable?

**Pull request merged**